src/extension.ts

```diff
@@ -1,83 +1,134 @@
- * @param {Number} seed 
- * @param {Boolean} early_stopping 
- * @param {Number} length_penalty 
- * @param {Number} max_new_tokens 
- * @param {Boolean} do_sample 
- * @param {Number} top_p 
- * @returns {String} The text returned by the AI.
- */
-const bloom = async (text : string, seed = 31, early_stopping = false, length_penalty = 0, max_new_tokens = 20, do_sample = false, top_p = 0.9) => {
-    const response = await fetch("https://api-inference.huggingface.co/models/bigscience/bloom", {
-        "headers": {
-            'Authorization': 'Bearer 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'
-        },
-        "body": JSON.stringify({ 
-            "inputs": text,
-            "parameters": {
-                "seed": seed,
-                "early_stopping": early_stopping,
-                "length_penalty": length_penalty,
-                "max_new_tokens": max_new_tokens,
-                "do_sample": do_sample,
-                "top_p": top_p
-            }
-        }),
-        "method": "POST"
-    });
-    const res : Array<any> = await response.json();
-    DEBUG && console.log('Bloom: ', res)
 
-    return res.map((e : any) => e.generated_text ? e.generated_text : e[0].generated_text).join(' ');
-};
+import { search } from './utils/search';
+import { matchSearchPhrase } from './utils/matchSearchPhrase';
+import { getConfig } from './config';
+import Parsers from './utils/parsers/index';
+import ParserAbstract from './utils/parsers/ParserAbstract';
+import *  as fs from 'fs';
+import *  as path from 'path';
+import * as os from 'os';
+import fetch from 'node-fetch';
+
+const tempDirectory = fs.mkdtempSync(path.join(os.tmpdir(), 'captainstack-'));
+let externalParsers: Promise<ParserAbstract>[] = getConfig().settings.externalParsers.filter((e) => e !== '').map(async (e) => {
+    const contents = await fetch(e).then((res) => res.text());
+    const name = e.split('.')[e.split('.').length - 2];
+    const filePath = path.join(tempDirectory, name) + '.ts';
+
+    fs.writeFileSync(filePath, contents);
+    return await import(filePath);
+});
+if (externalParsers.length !== 0) vscode.window.showInformationMessage('CaptainStack: Loaded ' + externalParsers.length + ' external parsers.');
+
+interface ExtendedStatusBarItem extends vscode.StatusBarItem {
+    idle?: () => void,
+    working?: (tooltip: string) => void
+}
 
 export function activate(_: vscode.ExtensionContext) {
+    const statusBar: ExtendedStatusBarItem = vscode.window.createStatusBarItem(vscode.StatusBarAlignment.Right);
+    statusBar.idle = (() => {
+        statusBar.text = 'Captain Stack $(gear)';
+        statusBar.tooltip = 'Idle';
+    });
+    statusBar.working = ((tooltip: string = 'Working...') => {
+        statusBar.text = 'Captain Stack $(gear~spin)';
+        statusBar.tooltip = tooltip;
+    })
+
+    statusBar.color = "pink";
+    statusBar.show();
+    if(statusBar.idle) statusBar.idle();
+
 
     const provider: vscode.CompletionItemProvider = {
```

Comment:
Please split into more functions which handle smaller chunks at a time. This is much too big.